src/intl/IntlLocale.cpp

```diff
@@ -368,5 +395,268 @@ IntlLocaleObject::IntlLocaleObject(ExecutionState& state, Object* proto, String*
     m_locale = localeBuilder.finalize();
 }
 
+Value IntlLocaleObject::calendars(ExecutionState& state)
+{
+    ValueVector resultVector;
+    if (m_calendar) {
+        resultVector.pushBack(m_calendar.value());
+    } else {
+        UErrorCode status = U_ZERO_ERROR;
+        LocalResourcePointer<UEnumeration> calendars(
+            ucal_getKeywordValuesForLocale("calendar", m_locale->toNonGCUTF8StringData().data(), true, &status),
+            [](UEnumeration* fmt) { uenum_close(fmt); });
+
+        if (!U_SUCCESS(status)) {
+            ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "Invalid locale");
+            return Value();
+        }
+
+        const char* buffer;
+        int32_t bufferLength = 0;
+        while ((buffer = uenum_next(calendars.get(), &bufferLength, &status)) && U_SUCCESS(status)) {
+            std::string calendar(buffer, bufferLength);
+            calendar = Intl::convertICUCalendarKeywordToBCP47KeywordIfNeeds(calendar);
+            resultVector.pushBack(String::fromUTF8(calendar.data(), calendar.size()));
+        }
+
+        if (!U_SUCCESS(status)) {
+            ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "Invalid locale");
+            return Value();
+        }
+    }
+
+    return Object::createArrayFromList(state, resultVector);
+}
+
+Value IntlLocaleObject::collations(ExecutionState& state)
+{
+    ValueVector resultVector;
+    if (m_collation) {
+        resultVector.pushBack(m_collation.value());
+    } else {
+        UErrorCode status = U_ZERO_ERROR;
```

Comment:
Same as above

Reply:
status used by ucol_getKeywordValuesForLocale

Reply:
um...I mean you can reduce unnecessary indentation same as above suggestion

Reply:
I cannot agree with your opinion.
I want to declare variable near place of what is used on
IMO distance is important than indentation